x-pack/plugins/session_view/common/utils/expand_dotted_object.test.ts

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { expandDottedObject } from './expand_dotted_object';
+
+const testFlattenedObj = {
+  'flattened.property.a': 'valueA',
+  'flattened.property.b': 'valueB',
+  regularProp: {
+    nestedProp: 'nestedValue',
+  },
+  'nested.array': [
+    {
+      arrayProp: 'arrayValue',
+    },
+  ],
+  emptyArray: [],
+};
+describe('expandDottedObject(obj)', () => {
+  it('works', () => {
```

Comment:
Nit: Grouping these together simplifies things up front, but makes it just a bit harder to debug if a subset of them fail. Could you separate them into `it('retrieves values from flattened keys', () => { ... });` and `it('retrieves values from nested keys', () => { ... })`?